toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1453,3 +1587,38 @@ func findAllRPMS(rpmsToFind []string) (foundAllRpms bool, missingRpms []string)
 
 	return
 }
+
+// findAllRPMSUpstream returns true if all RPMs requested are found in PMC/repo.
+//
+//	This is used to check if all RPMs are available in PMC/repo before starting a build.
+//	needs repo-list, manifest files rpmrepocloner
+func findAllRPMSUpstream(rpmsToFind []*PkgNode, ignoreVersionToResolveSelfDep bool, cloner *rpmrepocloner.RpmRepoCloner) (foundAllRpms bool, err error) {
+	/*assuming allRPMs found, to be set to false on error return or when a package is not found*/
+	foundAllRpms = true
+	for _, node := range rpmsToFind {
+		logger.Log.Debugf("Searching for a package which supplies: %s", node.VersionedPkg.Name)
+		// Resolve nodes to exact package names so they can be referenced in the graph.
+		if ignoreVersionToResolveSelfDep {
+			/*clear Version information so that any available version and release can be used*/
+			clearVersion(node.VersionedPkg)
+		}
+		//initialize an array of strings to hold the resolved packages
+		var resolvedPackages []string
+		resolvedPackages, err = cloner.WhatProvides(node.VersionedPkg)
```

Comment:
Didn't build this myself, but I'm fairly sure this will pollute the original node since its a pointer.
```suggestion
		// Resolve nodes to exact package names so they can be referenced in the graph.
		searchVersion := *node.VersionedPkg
		if ignoreVersionToResolveSelfDep {
			/*clear Version information so that any available version and release can be used*/
			clearVersion(searchVersion)
		}
		//initialize an array of strings to hold the resolved packages
		var resolvedPackages []string
		resolvedPackages, err = cloner.WhatProvides(searchVersion)
```